homeassistant/components/filesize/sensor.py

```diff
@@ -12,69 +12,76 @@
     PLATFORM_SCHEMA as PARENT_PLATFORM_SCHEMA,
     SensorEntity,
 )
-from homeassistant.const import DATA_MEGABYTES
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_FILE_PATH, DATA_MEGABYTES
 from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.reload import setup_reload_service
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import DOMAIN, PLATFORMS
+from .const import CONF_FILE_PATHS, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-
-CONF_FILE_PATHS = "file_paths"
 ICON = "mdi:file"
 
 PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
     {vol.Required(CONF_FILE_PATHS): vol.All(cv.ensure_list, [cv.isfile])}
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the file size sensor."""
 
-    setup_reload_service(hass, DOMAIN, PLATFORMS)
-
-    sensors = []
-    paths = set()
     for path in config[CONF_FILE_PATHS]:
-        try:
-            fullpath = str(pathlib.Path(path).absolute())
-        except OSError as error:
-            _LOGGER.error("Can not access file %s, error %s", path, error)
-            continue
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={CONF_FILE_PATH: path},
+            )
+        )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the platform from config entry."""
 
-        if fullpath in paths:
-            continue
-        paths.add(fullpath)
+    path = entry.data[CONF_FILE_PATH]
 
-        if not hass.config.is_allowed_path(path):
-            _LOGGER.error("Filepath %s is not valid or allowed", path)
-            continue
+    try:
+        get_path = await hass.async_add_executor_job(pathlib.Path, path)
+        fullpath = str(get_path.absolute())
+    except OSError as error:
+        _LOGGER.error("Can not access file %s, error %s", path, error)
+        return
 
-        sensors.append(Filesize(fullpath))
+    if not hass.config.is_allowed_path(path):
+        _LOGGER.error("Filepath %s is not valid or allowed", path)
+        return
```

Comment:
We should move this logic to the integration entry setup. That way, we can abort/fail the entry setup with an error message that shows in the UI.

The structure you are using right now, will cause no messages in the UI and solely results in missing entities.